pkg/config/config.go

```diff
@@ -19,38 +20,59 @@ type Config struct {
 		Git GitRepositories `glu:"git"`
 		OCI OCIRepositories `glu:"oci"`
 	} `glu:"sources"`
+	Server Server `glu:"server"`
 }
 
-func (c *Config) setDefaults() error {
-	if err := c.Log.setDefaults(); err != nil {
-		return err
-	}
-
-	if err := c.Sources.Git.setDefaults(); err != nil {
-		return err
-	}
+type defaulter interface {
+	setDefaults() error
+}
 
-	if err := c.Sources.OCI.setDefaults(); err != nil {
-		return err
-	}
+func (c *Config) SetDefaults() error {
+	return processStruct(reflect.ValueOf(c).Elem(), func(d defaulter) error {
+		return d.setDefaults()
+	})
+}
 
-	return nil
+type validater interface {
+	validate() error
 }
 
 func (c *Config) Validate() error {
-	if err := c.Log.validate(); err != nil {
-		return err
-	}
+	return processStruct(reflect.ValueOf(c).Elem(), func(v validater) error {
+		return v.validate()
+	})
+}
 
-	if err := c.Sources.Git.validate(); err != nil {
-		return err
-	}
+func processStruct[T any](val reflect.Value, method func(T) error) error {
+	switch val.Kind() {
+	case reflect.Struct:
+		// Need to get addressable value for pointer receiver methods
+		if val.CanAddr() {
+			if impl, ok := val.Addr().Interface().(T); ok {
+				if err := method(impl); err != nil {
+					return err
+				}
+			}
+		}
 
-	if err := c.Sources.OCI.validate(); err != nil {
-		return err
+		// Recursively process only struct fields
+		for i := 0; i < val.NumField(); i++ {
+			field := val.Field(i)
+			if field.Kind() == reflect.Struct ||
+				(field.Kind() == reflect.Ptr && !field.IsNil() && field.Elem().Kind() == reflect.Struct) {
+				if err := processStruct(field, method); err != nil {
+					return err
+				}
+			}
```

Comment:
You can drop these conditionals as it is handled implicitly by `processStruct` switch case.
```suggestion
			if err := processStruct(val.Field(i), method); err != nil {
				return err
			}
```